internal/statesync/syncer.go

```diff
@@ -161,7 +161,14 @@ func (s *syncer) SyncAny(discoveryTime time.Duration, retryHook func()) (sm.Stat
 	for {
 		// If not nil, we're going to retry restoration of the same snapshot.
 		if snapshot == nil {
-			snapshot = s.snapshots.Best()
+			if targetHeight > 0 {
+				snapshot, err = s.snapshots.At(uint64(targetHeight))
+				if err != nil {
```

Comment:
Not finding a snapshot is not a reason for leaving this method. As you can observe after line 174, we try again retrieving the snapshot (we send a further `SnapshotsRequest` to all peers) if, for instance, we didn't receive it yet.